src/App.tsx

```diff
@@ -101,6 +102,7 @@ function App() {
           setAp(portal);
           setNetwork(essid || "");
           setSerialNumber(serial);
+          setOsVersion(osVersion);
```

Comment:
You have to extract osVersion from the object in function parameter otherwise you are just setting always the same value (which is `""`) to the state so it does absolutely nothing. You also need to modify the API (in api.ts) accordingly (so it now returns the osVersion)